bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliserverBridgeHandler.java

```diff
@@ -0,0 +1,207 @@
+ */
+package org.openhab.binding.dali.internal.handler;
+
+import static org.openhab.binding.dali.internal.DaliBindingConstants.*;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.dali.internal.protocol.DaliBackwardFrame;
+import org.openhab.binding.dali.internal.protocol.DaliCommandBase;
+import org.openhab.binding.dali.internal.protocol.DaliResponse;
+import org.openhab.core.common.NamedThreadFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DaliserverBridgeHandler} handles the lifecycle of daliserver connections.
+ *
+ * @author Robert Schmid - Initial contribution
+ */
+@NonNullByDefault
+public class DaliserverBridgeHandler extends BaseBridgeHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(BRIDGE_TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(DaliserverBridgeHandler.class);
+    private static final int HEARTBEAT_MINUTES = 2;
+    private static final int DALI_DEFAULT_TIMEOUT = 5000;
+
+    private DaliserverConfig config = new DaliserverConfig();
+    private @Nullable ScheduledFuture<?> heartbeat;
+    private @Nullable ExecutorService commandExecutor;
+
+    public DaliserverBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing dali bridge");
+        config = getThing().getConfiguration().as(DaliserverConfig.class);
+
+        commandExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        if (validConfiguration(config)) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    private boolean validConfiguration(@Nullable DaliserverConfig config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "dali configuration missing");
+            return false;
+        }
+
+        if (config.host.isEmpty()) {
```

Comment:
This can not occur as you have required="true" defined in the XML, so if it is empty the handler will never run. This point is moot as I believe you can remove this whole function if you use the above suggestion.